tests/integration_tests/performance/test_block_performance.py

```diff
@@ -336,3 +341,56 @@ def test_block_performance(
         fio_block_size,
         vm,
     )
+
+
+def pin_backend(backend, cpu_id: int):
+    """Pin the vhost-user backend to a cpu list."""
+    return ProcessManager.set_cpu_affinity(backend.pid, [cpu_id])
+
+
+@pytest.mark.nonci
+@pytest.mark.timeout(RUNTIME_SEC * 1000)  # 1.40 hours
+@pytest.mark.parametrize("vcpus", [1, 2], ids=["1vcpu", "2vcpu"])
+@pytest.mark.parametrize("fio_mode", ["randread", "randwrite"])
+@pytest.mark.parametrize("fio_block_size", [4096], ids=["bs4096"])
+def test_block_vhost_user_performance(
+    microvm_factory,
+    guest_kernel,
+    rootfs,
+    vcpus,
+    fio_mode,
+    fio_block_size,
+    st_core,
+):
+    """
+    Execute block device emulation benchmarking scenarios.
+    """
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn(log_level="Info")
+    vm.basic_config(vcpu_count=vcpus, mem_size_mib=GUEST_MEM_MIB)
+    vm.add_net_iface()
+
+    # Add a secondary block device for benchmark tests.
+    fs = drive_tools.FilesystemFile(size=BLOCK_DEVICE_SIZE_MB)
+    backend = spawn_vhost_user_backend(vm, fs.path, readonly=False)
+    vm.add_vhost_user_block("scratch", VHOST_USER_SOCKET)
+    vm.start()
+
+    # Get names of threads in Firecracker.
+    current_cpu_id = 0
+    vm.pin_vmm(current_cpu_id)
+    current_cpu_id += 1
+    vm.pin_api(current_cpu_id)
+    pin_backend(backend, 2)
+    for vcpu_id in range(vm.vcpus_count):
+        current_cpu_id += 2
```

Comment:
shouldn't this be `+= 1`?

Reply:
indeed, I found a way to mess it up. Fixed, thanks!